src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,95 @@
+package core.basesyntax.service;
+
+import core.basesyntax.model.User;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static RegistrationService registrationService;
+
+    @Before
+    public void setUp() {
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @Test
+    public void testRegister_ValidUser_Success() throws UserRegistrationException {
+        RegistrationService registrationService = new RegistrationServiceImpl();
+        User user = new User();
+        user.setLogin("username");
+        user.setPassword("password");
+        user.setAge(20);
+        User registeredUser = registrationService.register(user);
+        Assert.assertEquals(user, registeredUser);
+    }
+
+    @Test
+    public void testRegister_UserWithExistingLogin_Failure() {
+        RegistrationService registrationService = new RegistrationServiceImpl();
+        User user1 = new User();
+        user1.setLogin("username");
+        user1.setPassword("password1");
+        user1.setAge(20);
+        User user2 = new User();
+        user2.setLogin("username");
+        user2.setPassword("password2");
+        user2.setAge(25);
+
+        try {
+            registrationService.register(user1);
+            registrationService.register(user2);
+            Assert.fail("Expected UserRegistrationException");
+        } catch (UserRegistrationException e) {
+            Assert.assertEquals("User with login username already exists.", e.getMessage());
+        }
+    }
+
+    @Test
+    public void testRegister_InvalidLoginLength_Failure() {
+        RegistrationService registrationService = new RegistrationServiceImpl();
+        User user = new User();
+        user.setLogin("user");
+        user.setPassword("password");
+        user.setAge(22);
```

Comment:
Please add creating default Ok user to Before method and in each test change only required filed (password to short one, etc). It will reduce code amount and make each test case clearer